### PR TITLE
[BugFix](sort) use correct agg function when using 2 phase sort for agg table

### PR DESCRIPTION
**Pull request by ByteYue**
# Proposed changes

Issue Number: close #xxx
The former logic would try to do 2 phase sort optimization for all table type. The newly added `__DORIS_ROWID_COL__` column in #15642 doesn't support agg function which would result coredump.
## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [x] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.42 seconds
 load time: 487 seconds
 storage size: 17122532887 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230129053307_clickbench_pr_86027.html

**Comment by eldenmoon** [comment]
we could modify __DORIS_ROWID_COL__ to `replace`

File: regression-test/suites/query_p0/sort/sort.groovy [hunk 1]
```diff
@@ -23,4 +23,22 @@ suite("sort") {
     qt_sort_string_single_column """ select * from ( select '汇总' as a union all select '2022-01-01' as a ) a order by 1 """
     qt_sort_string_multiple_columns """ select * from ( select '汇总' as a,1 as b union all select '2022-01-01' as a,1 as b ) a order by 1,2 """
     qt_sort_string_on_fe """ select '汇总' > '2022-01-01' """
+
+    sql """CREATE TABLE IF NOT EXISTS NO2PhaseSortWhenAggTable
```
Comment (eldenmoon) [inline regression-test/suites/query_p0/sort/sort.groovy:27]: `2PhaseSortWhenAggTable` could be more correct

Comment (ByteYue) [reply regression-test/suites/query_p0/sort/sort.groovy:27]: Oh, I'll change it soon.

**Review by eldenmoon — APPROVED** [review]
LGTM

**Review by eldenmoon — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**